apps/balancer-tools/src/app/apr/rounds/[roundId]/route.ts

```diff
@@ -1,15 +1,39 @@
 import { NextRequest, NextResponse } from "next/server";
 
+import { getBalEmissions } from "#/app/apr/(utils)/getBalEmission";
+import { getBALPriceByRound } from "#/app/apr/(utils)/getBALPriceByRound";
+import { Round } from "#/app/apr/(utils)/rounds";
+
 // import { DuneAPI } from "#/lib/dune";
-import { voteGaugeByID } from "../../mock_apis";
+import { mockGetTVLByRoundId, voteGaugeByID } from "../../mock_apis";
 
 export async function GET(
   _: NextRequest,
   context: { params: { roundId: number } },
 ) {
   if (!context.params.roundId) return "Missing round_id";
+  const selectedRound = Round.getRoundByNumber(context.params.roundId);
+
   // const duneAPI = new DuneAPI();
-  // const dune_request = await duneAPI.getPoolsByRoundId(context.params.roundId)
-  const dune_request = voteGaugeByID;
-  return NextResponse.json(dune_request);
+  // const requestJson = await duneAPI.getPoolsByRoundId(context.params.roundId)
+  const requestJson = voteGaugeByID;
+
+  // const totalTVLRequest = await duneAPI.getTVLByRoundId(context.params.roundId)
+  const totalTVLRequest = mockGetTVLByRoundId;
+  const totalTvl = totalTVLRequest[0]["total_tvl"];
+
+  const balPrice = await getBALPriceByRound(selectedRound);
+  const balWeeklyEmissions = getBalEmissions(
+    selectedRound.endDate.getFullYear(),
+  )["weekly"];
+
+  requestJson.forEach((poolData) => {
+    // APR = emissions in week * voting share * weeks in year * BAL price / TVL
+    poolData["apr"] =
+      (balWeeklyEmissions * poolData.pct_votes * 52 * balPrice) /
+      totalTvl /
+      100;
+    return poolData;
+  });
```

Comment:
do we even need an APR endpoint and that same logic implemented here?

Reply:
why are you plugging a random TVL to every pool in that list?

Reply:
> do we even need an APR endpoint and that same logic implemented here?

Not necessarily. I began developing it as its own route (which is why the files are present). However, considering that we would want to retrieve the APR when fetching the pools, I thought it might be more convenient to return everything within a single endpoint.

> why are you plugging a random TVL to every pool in that list?

That's another aspect I'm not entirely certain about how to approach. Querying for individual pools is inefficient, and querying for all pools is expensive. The only solution I can currently think of is to perform a query for a list (using SQL IN), but I haven't tested that yet.